### DISCUSSION TITLE
Has anyone been able to send an IPC event from the UI app back to NX Electron's main thread?

### DISCUSSION THREAD

**Discussion by mr-robot-o** [bennymeg/nx-electron#182] [Q&A] [answered]
I'm trying to find a way to send events back and forth between the UI app and main thread in NX electron, however I'm getting errors when trying to load Electron into my React frontend since it's setup as a web frontend.

Has anyone been able to send a message back and forth using ipcMain?

I see it being done strictly inside the electron app, however not sure if it can be done with the setup here.

**Comment by bennymeg** [comment, marked as answer]
I have opted to ship the default app with some security measures (which can be disabled if you wish).
Follow [this](https://www.electronjs.org/docs/latest/api/context-bridge/) article to define a context bridge (inside of the preload script) between the IPC main and the IPC renderer (there are some example functions defined as an example).

**Reply by mr-robot-o** [reply]
Thanks @bennymeg, just curious is this route considered less secure?

**Reply by bennymeg** [reply]
In this case, it will be disabling the context isolation and invoking the IPC renderer directly from the renderer process.
